explainer.md

```diff
@@ -0,0 +1,329 @@
+
+For example, an Android application developer may use the standard Android components,
+which will expose the correct semantics and bounding boxes for touch exploration,
+and have simple hooks for setting the accessible name.
+
+However, if they are creating a fully-customised user interface,
+the framework gives them the ability to create a
+[virtual view hierarchy](https://developer.android.com/guide/topics/ui/accessibility/apps.html#virtual-hierarchy),
+effectively exposing a sub-tree of accessibility nodes for a single, complex view.
+
+### Gaps in the web platform's accessibility story
+
+Since the web is missing this type of low-level API, it leads to significant gaps.
+Web apps that push the boundaries of what's possible on the web struggle to make them accessible
+because the APIs aren't yet sufficient.
+New web platform features aren't fully accessible
+or don't interact well with existing accessibility APIs,
+forcing developers to choose between using new standard or remaining accessible.
+
+* A library author creating a custom element is forced to "sprout" ARIA attributes
+to express semantics which are implicit for native elements.
+
+```html
+<!-- Page author uses the custom element as they would a native element -->
+<custom-slider min="0" max="5" value="3"></custom-slider>
+
+<!-- Custom element is forced to "sprout" extra attributes to express semantics -->
+<custom-slider min="0" max="5" value="3" role="slider"
+               tabindex="0" aria-valuemin="0" aria-valuemax="5"
+               aria-valuenow="3" aria-valuetext="3"></custom-slider>
+```
+
+* Moreover, there is no way to connect custom HTML elements to accessible actions.
+For example, the custom slider above with a role of `slider`
+prompts a suggestion on VoiceOver for iOS
+to perform swipe gestures to increment or decrement,
+but there is no way to handle that gesture via the DOM API.
+
+* Many ARIA relationship properties depend on IDREFs,
+meaning that elements participating in these relationship must have globally unique IDs.
+This is ugly, costly and fiddly to achieve,
+particularly in the context of framework or library code.
+
+* A custom element which uses shadow DOM
+and which needs to express semantic relationships such as
+["active descendant"](https://www.w3.org/TR/wai-aria-1.1/#aria-activedescendant)
+may be unable to do so, as `aria-activedescendant` relies on IDREF values,
+which are scoped to a single document fragment.
+
+```html
+<x-combobox>
+  #shadow-root
+  |  <!-- this doesn't work! -->
+  |  <input aria-activedescendant="opt1"></input>
+  <x-optionlist>
+    <x-option id="opt1">Option 1</x-option>
+    <x-option id="opt2">Option 2</x-option>
+    <x-option id='opt3'>Option 3</x-option>
+ </x-optionlist>
+</x-combobox>
+
+```
+
+
+* Custom UI which uses canvas to draw the UI currently 
+
+A low-level API would bridge these gap,
```

Comment:
a low-level API similar to what? This part felt a little hand wavy to me. Could maybe just say "a low-level API giving developers programmatic access to accessibility nodes would bridge these gaps" ?